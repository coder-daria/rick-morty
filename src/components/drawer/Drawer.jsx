import React, { useCallback } from 'react';
import { bool, node, string } from 'prop-types';
import { Drawer as AntdDrawer } from 'antd';
import { useReactiveVar } from '@apollo/client';

import { isDrawerOpenVar, selectedListItemVar } from '../../apollo/cache';

const DRAWER_PROPS = {
  PLACEMENT: 'right',
  CLOSABLE: true,
  WIDTH: '350px',
};

const { PLACEMENT, CLOSABLE, WIDTH } = DRAWER_PROPS;

function Drawer({ width, title, closable, placement, children }) {
  const isDrawerOpen = useReactiveVar(isDrawerOpenVar);

  const handleDrawerClose = useCallback(() => {
    isDrawerOpenVar(null);
    selectedListItemVar(0);
  }, []);

  return (
    <AntdDrawer
      closable={closable}
      onClose={handleDrawerClose}
      placement={placement}
      title={title}
      visible={isDrawerOpen}
      width={width}
    >
      {children}
    </AntdDrawer>
  );
}

Drawer.propTypes = {
  children: node.isRequired,
  closable: bool,
  placement: string,
  title: string.isRequired,
  width: string,
};

Drawer.defaultProps = {
  closable: CLOSABLE,
  placement: PLACEMENT,
  width: WIDTH,
};

export default Drawer;
