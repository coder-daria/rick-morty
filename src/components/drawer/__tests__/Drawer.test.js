import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Drawer from '../Drawer';

const setup = () => {
  const props = {
    title: 'Drawer title',
    children: <div>Drawer content</div>,
  };

  // eslint-disable-next-line react/jsx-props-no-spreading
  return shallow(<Drawer {...props} />);
};

describe('<Drawer />', () => {
  it('should match the snapshot', () => {
    const wrapper = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
