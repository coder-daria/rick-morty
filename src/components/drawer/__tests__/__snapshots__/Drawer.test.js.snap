// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> should match the snapshot 1`] = `
<DrawerWrapper
  closable={true}
  onClose={[Function]}
  placement="right"
  title="Drawer title"
  visible={false}
  width="350px"
>
  <div>
    Drawer content
  </div>
</DrawerWrapper>
`;
